experiments/modern/src/runtime/MakiObject.js

```diff
@@ -3,25 +3,18 @@ import { findElementById, findGroupDefById } from "../utils";
 
 class MakiObject {
   constructor(node, parent, annotations = {}) {
-    this.xmlNode = node;
+    if (node) {
+      this.attributes = node.attributes;
+      this.name = node.name;
```

Comment:
Exposing these seems basically the same as exposing the xml node itself. Do you think we can get to a place where we expose the individual properties/attributes as explicit properties rather than implicitly inheriting from whatever XML we get?

Also, again we're exposing js properties without a `js_` prefix. I worry that it's going to get really hard to tell what's our code that we can play with and what's there for the sake of the Maki scripts.

Any other ideas how we could separate the two kinds of properties/methods?

All that said, this is still a net positive, so I'm happy to merge it.

Reply:
It does seem like many of the methods exist on `GuiObject` so its probably possible to start using them more to limit the direct access of `attributes`

Reply:
gonna think about it more

Reply:
Thought about it, but no epiphanies unfortunately :( Maybe as we start implementing more and more properties things will become a bit clearer.

Rebased and gonna merge for now